spec/std/system_spec.cr

```diff
@@ -18,4 +18,12 @@ describe System do
       cpu_count.should eq(shell_cpus)
     end
   end
+
+  describe "login" do
+    it "returns String when found" do
+      shell_login = `echo $USER`.strip
```

Comment:
No, it seems correct. `$USER` is `""` or unset, and `System.login` is `Nil`. It seems the bug is in the spec for not equating `""` and `Nil`. It also just occured to me that `` `echo $USER`.strip `` is a weird way to get the value when we have `ENV["USER"]`. Using `ENV["USER"]?` would give the correct behaviour of being `Nil` when unset.

The login name is `Nil` because the specs are running inside a docker container - which has no login session. That brings us to another question, why is this method useful? It doesn't tell us the current effective user of the process, just the login session.